WLED Audio

netmindz opened the pull request.

netmindz committed:
Add usermod to receive WLED Audio sync data

netmindz committed:
Add usermod to receive WLED Audio sync data

netmindz committed:
Copy paste of WLEDs GEQ

netmindz committed:
Start updating GEQ effect for StarMod

netmindz committed:
Disable debug print statements

ewoudwijma committed:
Change GEQEffect to make it compile (WIP)

netmindz commented on src/App/AppEffects.h:
```diff
@@ -223,39 +224,45 @@ class Frizzles2D:public Effect {
   }
 }; // Frizzles2D
 
+#define NUM_GEQ_CHANNELS 16                                           // number of frequency channels. Don't change !!
```
Should be getter from wledAudioMod

ewoudwijma replied:
Agree

netmindz commented on src/App/AppEffects.h:
```diff
@@ -223,39 +224,45 @@ class Frizzles2D:public Effect {
+  byte previousBarHeight[1024];
+  uint8_t intensity;
+  uint8_t speed;
+  bool check1;
+  bool check2;
```
Should be named as per their function in the effect, not based on WLED param names

ewoudwijma replied:
This is quick and dirty, I was thinking of a function like getControls which calls initSlider(..., control name, ...)

Similar as now done in fx.chFun

This function does not claim room of variables in the class.

Only drawback is that the effect itself should call getValue("control name") which is too slow, so maybe store that value during setup(), but then again we create local class variables...

netmindz commented on src/App/AppEffects.h:
```diff
@@ -223,39 +224,45 @@ class Frizzles2D:public Effect {
 
-    if (SEGENV.call == 0) SEGMENT.fill(BLACK);
-    int fadeoutDelay = (256 - SEGMENT.speed) / 64;
-    if ((fadeoutDelay <= 1 ) || ((SEGENV.call % fadeoutDelay) == 0)) SEGMENT.fadeToBlackBy(SEGMENT.speed);
```
To make porting easier and generally code nicer - should we not just a have  `fadeToBlackBy` method that takes only speed (assuming you are controlling all the current LEDs)

ewoudwijma replied:
Maybe, don't know yet, I now converted them to the fastled version, which I like also (the fastled compatibility)

netmindz replied:
This would just be wrapper for the fastled one, just more convenient to not pass the same two params around

netmindz committed:
Merge branch 'main' into Audio

netmindz committed:
Fix compile errors

netmindz committed:
Disable WLEDAudio debug

netmindz committed:
Setting speed and intensity from e131

netmindz committed:
Setting speed and intensity from model rather than direct from e131

netmindz committed:
Comment now probably redudant getValue from e131 mod

ewoudwijma committed:
Implement Effect controls

AppEffects:
- refactor parameters function
- use parameters in GEQ

AppModLeds
- fx.chFun: check if no bogus fx, get the right effect and perform setup() NEW and parameters, if no effect found then still show the random controls

ewoudwijma committed:
Merge remote-tracking branch 'netmindz/Audio' into Audio

ewoudwijma committed:
Fix

netmindz committed:
Map DMX values to the range of the variable, rather than using mod

netmindz committed:
Move e131 channel assigment to effect

netmindz committed:
Rename addWatch to patchChannel to better match its actual function of mapping dmx channel to a variable

netmindz committed:
Rename addWatch to patchChannel to better match its actual function of mapping dmx channel to a variable

netmindz committed:
Add AudioRings effect

netmindz committed:
Add comment about ring ordering

netmindz committed:
Add base class for 241Ring effects and add RandomFlow as example of non-audio

netmindz committed:
Reset "old" value when patching

netmindz committed:
Reset "old" value when patching

netmindz committed:
Merge branch 'main' into Audio

netmindz committed:
Swap to expecting mapping to be in LEDFixture

netmindz committed:
Merge branch 'ewowi:main' into Audio

ewowi committed:
Merge branch 'main' into Audio

ewowi committed:
Post merge

AppEffects: 
- add 1D/2D to name (temporary)
- GEQ 2D rename check1/check2
- rename parameters to controls

UserModWLEDAudio
- default not ebabled
- use onOffChanged

ewowi commented:
Merged latest main into audio and did some post merge changes.
Expected a not of this in this PR but apparently not ...

netmindz commented on src/App/AppEffects.h:
```diff
@@ -614,7 +614,7 @@ class AudioRings:public RingEffect {
 
     void loop() {
-      for (int i = 0; i < 7; i++) {
+      for (int i = 0; i < 7; i++) { // 7 rings
```
Actually 7 EQ bands as was written for MSGEQ7 first
